File: leetcode/3206-alternating-groups-i/numberOfAlternatingGroups_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type NumberOfAlternatingGroupsCase struct {
	Input []int

	Output int
}

func TestNumberOfAlternatingGroups(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]NumberOfAlternatingGroupsCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := numberOfAlternatingGroups(testCase.Input)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/3206-alternating-groups-i/numberOfAlternatingGroups.test.ts
import cases from "./cases.json";
import numberOfAlternatingGroups from "./numberOfAlternatingGroups";

test("alternating groups i", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    console.time("time");
    const result = numberOfAlternatingGroups(testCase.input);
    console.timeEnd("time");
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});
